.github/workflows/ci.yml

```diff
@@ -126,14 +137,28 @@ jobs:
         uses: actions/setup-go@v4
         with:
           go-version: ${{ env.GO_VERSION }}
+          cache-dependency-path: |
+            go.sum
+            test/go.sum
+
+      - name: Pre-fill Module Cache
+        shell: powershell
+        run: |
+          go mod download
+          cd test
+          go mod download
 
       - name: Validate go.mod and vendoring
         shell: powershell
         run: |
           Write-Output "::group::go mod tidy & vendor"
-          go mod tidy
-          go mod vendor
+          go mod tidy -v -e
+          go mod vendor -e
           Write-Output "::endgroup::"
+          if ($LASTEXITCODE -ne 0) {
+            Write-Output "::error title=Go Mod::Error running ``go mod tidy``` or ``go mod vendor``"
+            exit $LASTEXITCODE
```

Comment:
Should we log the actual exit code here or is it already logged by the system at a easy-to-see place?

Reply:
since we exit with it, github CI will say "exited with <#>", so we can view it there